# EXPLORATORY DATA ANALYSIS - NUMERIC

- The purpose of this task is to understand how to go about analysing data, what insights you can generate and any recommendations/Next steps to suggest.
- Is there an area of the website or business that the company can improve?

# 1. Basic Exploratory Analysis

### - Importing the necessary libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.graph_objs as go
from plotly.offline import init_notebook_mode, iplot
import itertools
%matplotlib inline

### - Reading the data:

In [2]:
df = pd.read_csv("Numerist Data Challenge May 2022.csv")
df.head()

Landing Page  Month of Year Sessions New Users  \
0           /global/country3/region2         201707  169,558    72,883   
1                  /global/country11         201701  132,740    68,161   
2                              /info         201701  117,876    70,112   
3  /global/country23/region3/resort3         201701  108,534    54,041   
4                  /global/country11         201707  104,034      Null   

  Bounces Pageviews Avg. Session Duration Transactions  
0  24,247   842,703               0:04:25          217  
1  38,946   670,337               0:04:22           98  
2  30,695   614,134               0:04:53          133  
3  25,343   539,414               0:04:37          251  
4  15,147   555,542               0:04:56           96

### - Exploring & cleaning the dataset

In [3]:
df.head(20)

Landing Page  Month of Year Sessions New Users  \
0            /global/country3/region2         201707  169,558    72,883   
1                   /global/country11         201701  132,740    68,161   
2                               /info         201701  117,876    70,112   
3   /global/country23/region3/resort3         201701  108,534    54,041   
4                   /global/country11         201707  104,034      Null   
5                   /global/country23         201707  101,576    49,510   
6                   /global/country11         201705  100,745    49,962   
7                   /global/country11         201702  100,591    50,062   
8            /global/country3/region2         201706   98,910    42,715   
9            /global/country3/region2         201705   95,860    42,516   
10  /global/country23/region3/resort3         201705   93,639    43,180   
11  /global/country23/region3/resort3         201707   90,908    40,614   
12           /global/country3/region2         201708   90,870    39,004   
13                  /global/country17         201704   87,658    44,186   
14                  /global/country23         201705   87,632    43,502   
15                  /global/country17         201707   87,593    41,315   
16           /global/country3/region2         201704   86,881    38,846   
17                  /global/country17         201705   85,199    41,276   
18                  /global/country23         201701   84,453    45,492   
19                  /global/country23         201706   84,062    41,030   

   Bounces Pageviews Avg. Session Duration Transactions  
0   24,247   842,703               0:04:25          217  
1   38,946   670,337               0:04:22           98  
2   30,695   614,134               0:04:53          133  
3   25,343   539,414               0:04:37          251  
4   15,147   555,542               0:04:56           96  
5   14,952   537,337               0:05:16          166  
6   15,243   541,001               0:05:09           94  
7   29,594   467,748               0:04:31           74  
8   14,955   478,724               0:04:38          156  
9   15,146   477,383               0:04:58          184  
10  11,724   452,276               0:04:32          209  
11  11,436   434,540               0:04:23          170  
12  12,622   457,076               0:04:30          131  
13  16,927   426,018               0:04:13          166  
14  12,978   472,336               0:05:28          120  
15  13,761   425,702               0:03:58          135  
16  15,447   443,093               0:05:06          186  
17  15,523   409,807               0:04:05          150  
18  21,155   445,912               0:05:25          126  
19  12,483   433,760               0:05:13          109

In [9]:
df.shape

(2441, 8)

- The data has 2441 rows and 8 columns
- This can be seen by the .tail() function

In [4]:
df.tail()

Landing Page  Month of Year Sessions  \
2436         /global/country14/region1/resort3         201710      789   
2437         /global/country23/region4/resort4         201708      789   
2438         /global/country26/region2/resort1         201708      789   
2439  /product/country1/region2/resort16/skew4         201708      789   
2440  /product/country11/region2/resort1/skew1         201703      789   

     New Users Bounces Pageviews Avg. Session Duration Transactions  
2436       476     170     3,101               0:02:50            0  
2437       355     194     3,590               0:04:44            0  
2438       345     100     4,040               0:04:00            2  
2439       460     238     3,101               0:03:20            2  
2440       339     175     4,268               0:04:21            0

In [5]:
df.columns

Index(['Landing Page', 'Month of Year', 'Sessions', 'New Users', 'Bounces',
       'Pageviews', 'Avg. Session Duration', 'Transactions'],
      dtype='object')

In [6]:
df.describe()

Month of Year
count    2441.000000
mean   201705.463335
std         2.798490
min    201701.000000
25%    201703.000000
50%    201706.000000
75%    201708.000000
max    201710.000000

- From the dataframe, I can conclude that:
    - The data is based on 2441 entries in a certain year
    - The entries are based between January 2017 and October 2017
    - The mean average sales in May was very high

In [10]:
#checking for the missing values in the data:
total = df.isnull().sum().sort_values(ascending = False)
percent = (df.isnull().sum()/df.isnull().count()*100).sort_values(ascending = False)
pd.concat([total, percent], axis = 1, keys = ["Total", "Percent"]).transpose()

Transactions  Avg. Session Duration  Pageviews  Bounces  New Users  \
Total             0.0                    0.0        0.0      0.0        0.0   
Percent           0.0                    0.0        0.0      0.0        0.0   

         Sessions  Month of Year  Landing Page  
Total         0.0            0.0           0.0  
Percent       0.0            0.0           0.0

- The dataset is clean and there is no missing values

# Observations at first sight

- The data seems to be that of a website.
- It shows the following details:
    - Landing Page.
    - Month of the year - Month of the year in which the dataset is from
    - Sessions - Probable sessions of a user
    - New Users - Number of new users visiting the page
    - Bounces - How many users bounced off the page
    - Pageviews - How many views did the page receive
    - Average session duration -The average time of a session for a user
    - Transactions - Number of transactions from a user on the page
- A landing page is a standalone web page that a person "lands" on after clicking through from an email, ad, or other digital location. Once they're on your landing page, users are encouraged to take an action, such as joining your list or buying your products.
- Definition derived from (https://mailchimp.com/marketing-glossary/landing-pages/#:~:text=A%20landing%20page%20is%20a%20standalone%20web%20page%20that%20a,list%20or%20buying%20your%20products.)

- From the dataset, you can already tell that the data is from a business webpage, evidenced by the "Average Session" that a user spends time on the site and the "Transactions" column.
- I can gather that the website was formed in 2017 and the data comes from the users in different countries.

## Questions I will be tackling:
    - Where do most users of the webpage come from?
    - Which month of the year got the most transactions?
    - What is the probability that someone will make a purchase based on the average time they spent on the website?
    - What is the average amount of bounces on the webpage?
    - What is the average number of pageviews?

# 2. Visualizations

- I will attempt to use visualizations found in the matplotlib and seaborn libraries of python to be able to accurately visualize the data
- Each question will be accompanied by its own visualization.
- The visualizations will be randomized and will follow no specific order

## -What is the average number of pageviews?

- #I intend to rename the entry in the column "Month of the year" from 201701 to read "01"
- #Then plot the month of the year vs pageviews using a bar graph to visualize the data 
#(23/05/2022)

### - Where do most users of the webpage come from?

In [11]:
#isolating the category of landing page to see where most consumers of the page come from:
df[df["Landing Page"] == "/global/country3"]

Landing Page  Month of Year Sessions New Users Bounces Pageviews  \
1146  /global/country3         201701    2,331       903     854    11,562   
1230  /global/country3         201708    2,069       856     426    10,717   
1272  /global/country3         201705    1,931       822     433    10,022   
1274  /global/country3         201707    1,922       832     422    10,302   
1291  /global/country3         201706    1,892       849     465     9,384   
1382  /global/country3         201704    1,685       770     405     8,931   
1431  /global/country3         201702    1,605       609     606     7,800   
1564  /global/country3         201703    1,403       569     369     7,197   
1813  /global/country3         201709    1,139       511     313     5,479   

     Avg. Session Duration Transactions  
1146               0:05:47            8  
1230               0:05:51            2  
1272               0:06:08            4  
1274               0:06:15            1  
1291               0:05:18            1  
1382               0:06:23            5  
1431               0:05:58            3  
1564               0:06:21            3  
1813               0:05:33            2

In [12]:
df[df["Landing Page"] == "/global/country11"]

Landing Page  Month of Year Sessions New Users Bounces Pageviews  \
1    /global/country11         201701  132,740    68,161  38,946   670,337   
4    /global/country11         201707  104,034      Null  15,147   555,542   
6    /global/country11         201705  100,745    49,962  15,243   541,001   
7    /global/country11         201702  100,591    50,062  29,594   467,748   
24   /global/country11         201706   80,953    39,630  12,807   417,717   
26   /global/country11         201703   79,553    39,952  17,740   405,720   
33   /global/country11         201704   73,864    36,739  12,254   401,820   
50   /global/country11         201708   60,501    29,834   9,178   336,991   
67   /global/country11         201702  100,591    50,062  29,594   467,748   
130  /global/country11         201709   31,531    16,072   5,531   170,898   
174  /global/country11         201710   23,286    12,339   4,606   120,389   

    Avg. Session Duration Transactions  
1                 0:04:22           98  
4                 0:04:56           96  
6                 0:05:09           94  
7                 0:04:31           74  
24                0:05:00           93  
26                0:04:57           84  
33                0:05:19           91  
50                0:05:16           59  
67                0:04:31           74  
130               0:05:20           39  
174               0:05:03           22

In [13]:
df[df["Landing Page"] == "/global/country23"]

Landing Page  Month of Year Sessions New Users Bounces Pageviews  \
5     /global/country23         201707  101,576    49,510  14,952   537,337   
14    /global/country23         201705   87,632    43,502  12,978   472,336   
18    /global/country23         201701   84,453    45,492  21,155   445,912   
19    /global/country23         201706   84,062    41,030  12,483   433,760   
25    /global/country23         201704   79,918    41,115  12,028   448,340   
34    /global/country23         201708   73,230    35,249  10,106   401,300   
43    /global/country23         201703   69,069    35,778  12,122   373,663   
57    /global/country23         201702   56,931    29,757  12,314   290,917   
105   /global/country23         201709   38,744    19,210   5,583   214,254   
137   /global/country23         201710   29,929    15,653   4,753   159,522   
2156  /global/country23         201701      923       477     338     3,563   

     Avg. Session Duration Transactions  
5                  0:05:16          166  
14                 0:05:28          120  
18                 0:05:25          126  
19                 0:05:13          109  
25                 0:05:49          153  
34                 0:05:24          110  
43                 0:05:47          136  
57                 0:05:36           88  
105                0:05:35           90  
137                0:05:11           55  
2156               0:03:03            1